pulser-core/pulser/sequence/sequence.py

```diff
@@ -2172,12 +2196,42 @@ def _validate_and_adjust_pulse(
         self, pulse: Pulse, channel: str, phase_ref: Optional[float] = None
     ) -> Pulse:
         channel_obj: Channel
+        detuning_map: DetuningMap | None = None
         if channel in self._schedule:
             channel_obj = self._schedule[channel].channel_obj
+            if isinstance(channel_obj, DMM):
+                detuning_map = cast(
+                    _DMMSchedule, self._schedule[channel]
+                ).detuning_map
         else:
             # Sequence is parametrized and channel is a dmm_name
-            channel_obj = self.device.dmm_channels[_dmm_id_from_name(channel)]
-        channel_obj.validate_pulse(pulse)
+            dmm_id = _dmm_id_from_name(channel)
+            channel_obj = self.device.dmm_channels[dmm_id]
+            dmm_idx = -1
+            for call in self._calls[1:] + self._to_build_calls:
+                if (
+                    call.name == "config_detuning_map"
+                    or call.name == "config_slm_mask"
+                ):
+                    # Check whether dmm_name matches with channel
```

Comment:
It seems that nothing happens if there is no match though, is that intended?

Reply:
Is it really okay that we are failing silently?

Reply:
It's would say it's not supposed to fail: the channel has been tested by `validate_channel` before being as being in `declared_channels`, and `declared_channels` is exactly built by looping over the calls.

Reply:
Ok, good enough for me

Reply:
Actually, let's add an assert in that case to ensure the detuning map is always found